src/core/ucc_lib.c

```diff
@@ -19,13 +21,235 @@ static ucc_config_field_t ucc_lib_config_table[] = {
+        ((params.coll_types & supported_coll_types) != params.coll_types)) {
+        ucc_debug("selected set of CLs does not provide all the requested "
+                  "coll_types");
+    }
+    if (params.thread_mode > supported_tm) {
+        ucc_debug("selected set of CLs does not provide the requested "
+                  "thread_mode");
+    }
+    lib->attr.coll_types  = supported_coll_types;
+    lib->attr.thread_mode = ucc_min(supported_tm, params.thread_mode);
+    return UCS_OK;
+
+error_cl_init:
+    ucc_config_parser_release_opts(cl_config, cl_iface->cl_lib_config.table);
+error_cl_config_parse:
+    free(cl_config);
+error_cl_cleanup:
+    for (i = 0; i < lib->n_libs_opened; i++) {
+        lib->libs[i]->iface->finalize(lib->libs[i]);
+    }
+error:
+    if (lib->libs)
+        free(lib->libs);
+    return status;
+}
+
 ucc_status_t ucc_init_version(unsigned api_major_version,
                               unsigned api_minor_version,
                               const ucc_lib_params_t *params,
-                              const ucc_lib_config_h *config,
-                              ucc_lib_h *lib_p)
+                              const ucc_lib_config_h config, ucc_lib_h *lib_p)
 {
-    return UCC_ERR_NOT_IMPLEMENTED;
+    unsigned        major_version, minor_version, release_number;
+    ucc_status_t    status;
+    ucc_lib_info_t *lib;
+
+    *lib_p = NULL;
+
+    if (UCC_OK != (status = ucc_constructor())) {
+        return status;
+    }
+
+    ucc_get_version(&major_version, &minor_version, &release_number);
+
+    if ((api_major_version != major_version) ||
+        ((api_major_version == major_version) &&
+         (api_minor_version > minor_version))) {
+        ucc_warn(
+            "UCC version is incompatible, required: %d.%d, actual: %d.%d.%d",
+            api_major_version, api_minor_version, major_version, minor_version,
+            release_number);
+    }
+
+    if (0 == ucc_global_config.cl_framework.n_components) {
+        ucc_error("no components are available in cl framework");
+        return UCC_ERR_NOT_FOUND;
+    }
+
+    lib = ucc_malloc(sizeof(ucc_lib_info_t), "lib_info");
+    if (!lib) {
+        ucc_error("failed to allocated %zd bytes for lib_info",
+                  sizeof(ucc_lib_info_t));
+        return UCC_ERR_NO_MEMORY;
+    }
+    lib->full_prefix = strdup(config->full_prefix);
+    if (!lib->full_prefix) {
+        ucc_error("failed strdup for full_prefix");
+        status = UCC_ERR_NO_MEMORY;
+        goto error;
+    }
+    /* Initialize ucc lib handle using requirements from the user
+       provided via params/config and available CLs in the
+       CL component framework.
+
+       The lib_p object will contain the array of ucc_cl_lib_t objects
+       that are allocated using CL init/finalize interface. */
+    status = ucc_lib_init_filtered(params, config, lib);
+    if (UCC_OK != status) {
+        goto error;
+    }
+    if (lib->n_libs_opened == 0) {
```

Comment:
Move this check to ucc_lib_init_filtered?

Reply:
done